src/provider_thruster/provider_thruster_node.cc

```diff
@@ -24,6 +24,20 @@ namespace provider_thruster {
         
 
         this->rs485Publisher = nh->advertise<sonia_common::SendRS485Msg>("/interface_rs485/dataRx", 1000);
+
+        const char* auv = std::getenv("AUV");
+
+        if (strcmp(auv, "AUV8") == 0){
+            SLAVE = sonia_common::SendRS485Msg::SLAVE_ESC;    
+        }
+
+        else if (strcmp(auv, "AUV7") == 0){
+            SLAVE = sonia_common::SendRS485Msg::SLAVE_PWR_MANAGEMENT;
+        }
+
+        else if (strcmp(auv, "LOCAL") == 0){
+            SLAVE = sonia_common::SendRS485Msg::SLAVE_PWR_MANAGEMENT;
+        }
```

Comment:
I would add a default so if no variable are there we fall back on something. See suggestion
```suggestion
        }
        else
        {
        ROS_ERROR_STREAM("No environnement variable found, using the default from AUV8");
        SALVE = sonia_common::SendRS485Msg::SLAVE_PWR_MANAGEMENT;
        }
```

Reply:
if no variable slave = SLAVE_ESC for auv8 not SLAVE_POWER_MANAGEMENT am I right ?